### ISSUE TITLE
[Feature Request] A way to provide initial guesses (x0) to fit_gpytorch_model / fit_gpytorch_mll

### ISSUE THREAD

**Issue by wjaekim** [pytorch/botorch#2103]
# 🚀 Feature Request

Ability to supply initial guesses of hyperparameters to the optimization routine of fit_gpytorch_model / fit_gpytorch_mll.

## Motivation

**Is your feature request related to a problem? Please describe.**

- I occasionally see an optimization result of fit_gpytorch_model / fit_gpytorch_mll is a local minimum, which is inevitable in all optimization problems.
- However, in sequential experimentation, it is a natural option to supply the preceding optimization result when optimizing the mll in the next experiment.
- Currently, BoTorch's optimization routine of the mll does not support user's ability to supply initial guesses.

I wonder if there is any trivial way to supply initial guesses but I am not aware? If so, please let me know.

**Comment by saitcakmak** [comment]
Hi @wjaekim. There isn't an explicit way to do this in the `fit_gpytorch_mll` API. However, since the first attempt at model fitting starts from the existing values of the hyper-parameters (if first attempt fails, it'll retry with random starting values), you can pass initial guesses by setting them on the model. An easy way to do this is `new_model.load_state_dict(old_model.state_dict())`. This will transfer all parameters & buffers of the old model to the new one.

**Comment by wjaekim** [comment]
Thank you so much for your suggestion. Loading the state_dict() this way won't replace the data---i.e., it will replace only the parameters & related buffers. Correct?

I just tried the line. I seems that it replaces only the parameter-related state.

Will implement the suggestion!

**Comment by saitcakmak** [comment]
Correct, this should not affect the data

**Comment by wjaekim** [comment]
Thanks. I found that I missed the [right document](https://pytorch.org/tutorials/beginner/saving_loading_models.html) for me.

It seems that to resume training with a saved state, I need to make sure it is set to training mode after loading it, by calling model.train(). Is this appropriate? Just to make sure.

With this information, I believe I am good to go.

**Comment by saitcakmak** [comment]
`fit_gpytorch_mll` will set the model to `train` mode before fitting. That'd only be necessary if you were to write a custom fitting loop. Calling `new_model.load_state_dict(old_model.state_dict())` before calling `fit_gpytorch_mll` should be all you need.

**Comment by wjaekim** [comment]
Got it. Then, that is everything I need. I truly appreciate your assistance. Saved my day!
